magellan-library/src/main/java/com/wealthfront/magellan/lifecycle/LifecycleTransitionExtensions.kt

```diff
@@ -0,0 +1,77 @@
+package com.wealthfront.magellan.lifecycle
+
+import android.content.Context
+import com.wealthfront.magellan.lifecycle.LifecycleState.Created
+import com.wealthfront.magellan.lifecycle.LifecycleState.Destroyed
+import com.wealthfront.magellan.lifecycle.LifecycleState.Resumed
+import com.wealthfront.magellan.lifecycle.LifecycleState.Shown
+import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.BACKWARDS
+import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.FORWARD
+import com.wealthfront.magellan.lifecycle.LifecycleStateDirection.NO_MOVEMENT
+
+public fun <T> T.transitionToState(newState: LifecycleState) where T : LifecycleAware, T : LifecycleOwner {
```

Comment:
See, generics can be fun 😉

I named it differently from `transition` below because I don't want people to get them confused, but I'm open to suggestions